### DISCUSSION TITLE
Nuxt 3 Middleware does not work when use one component for two routes

### DISCUSSION THREAD

**Discussion by JuliWolf** [nuxt/nuxt#28539] [Questions] [answered]
Hello

I have two routes which use the same component

```
  {
    path: '/first_page/',
    file: '~/pages/FirstPage.vue',
    name: 'firstRoute',
    meta: { header: 'firstRoute' },
  },
  {
    path: '/second_page/',
    file: '~/pages/FirstPage.vue',
    name: 'secondRoute',
    meta: { header: 'secondRoute' },
  }
```

This component has list of middlewares to run when page opens 4but `definePageMeta` works only for first page in route list
Second page always opens on default layout and middlewares not triggered

Reproduction https://stackblitz.com/edit/nuxt-starter-kwjxwg?file=middleware%2Ftest-two.ts

I also tried to setup page like this
This variant does not change anything

```
<script lang="ts">
import FirstPage from '~/pages/FirstPage.vue'

export default FirstPage
</script>
```

**Comment by AloisSeckar** [comment]
I would personally create two separate page components to avoid such issues and hoist their common content into `~/components/PageContent.vue`

**Comment by xjccc** [comment, marked as answer]
I think you could set a key for definePageMeta.

```
definePageMeta({
    key: route => route.name
})
```

And in nuxt.config.ts

```
experimental: {
  scanPageMeta: false
}
```

Mybe something problem with `scanPageMeta` for key. https://github.com/nuxt/nuxt/issues/28086

**Reply by JuliWolf** [reply]
This works for me. Thank you
